File: Java/LongestSubbaray_withZeroSum.java
import java.util.HashMap;

public class Solution {
    public static int getLongestZeroSumSubarrayLength(int []arr){
        HashMap<Integer, Integer> hmap = new HashMap<>();
        int sum = 0;
        int max_len = 0;

        for(int i = 0; i < arr.length; i++){
            sum += arr[i];

            if (sum == 0){
                max_len = Math.max(max_len, i+1);
            }
            else if(hmap.containsKey(sum)){
                max_len = Math.max(max_len, i - hmap.get(sum));
            }
            else{
                hmap.put(sum, i);
            }
        }
        return max_len;
    }
}

File: CPP/LongestSubbaray_with_ZeroSum.cpp
#include<vector>
using namespace std;
int getLongestZeroSumSubarrayLength(vector<int> &arr){
	unordered_map <int, int> hmap;
	int sum = 0, maxLen = 0;

	for(int i = 0; i < arr.size(); ++i){
		sum += arr[i];

                if (sum == 0) {
                  maxLen = max(maxLen, i + 1);
                } 
				else if (hmap.find(sum) != hmap.end()) {
                  maxLen = std::max(maxLen, i - hmap[sum]);
                } else {
                  hmap[sum] = i;
			}
        }
	return maxLen;
	
}
